packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -721,14 +719,13 @@ export const makeInnerVault = (
 
     // for status/debugging
     getCollateralAmount,
-    getDebtAmount,
+    getCurrentDebt,
     getNormalizedDebt,
     getLiquidationSeat: () => liquidationSeat,
   });
 
   return innerVault;
 };
 
-/**
- * @typedef {ReturnType<typeof makeInnerVault>} InnerVault
- */
+/** @typedef {ReturnType<typeof makeInnerVault>} InnerVault */
+/** @typedef {Unpromise<ReturnType<InnerVault['initVaultKit']>>} VaultKit */
```

Comment:
I haven't seen `Unpromise` before. What does it mean?

and how does `ReturnType` work? These two lines seem to be using it in different ways.  Why doesn't the return type of `makeInnerVault` just go on its declaration?

Reply:
> I haven't seen `Unpromise` before. What does it mean?

It's a handy type @michaelfig defined for us in https://github.com/Agoric/agoric-sdk/commit/20941e675302ee5905e4825638e661065ad5d3f9#diff-113f01ea0b5f91ff269bcc4b0a0b5f8391d8efc186eeb13197086dd9ad39aabcR8 . It unwraps a promise type to what it promises to return.

> and how does `ReturnType` work?

It evaluates to the return type of the argument (which must be a function type).

> These two lines seem to be using it in different ways.

Is the difference you're referring to the use of `typeof`? That's because `makeInnerVault` is a value, not a type. It needs the `typeof` to extract the type from the value. `InnerVault` is already a type.

> Why doesn't the return type of `makeInnerVault` just go on its declaration?

It would be nice to have a way to jump there. I figure it doesn't because the type just happens to be a return type. The normal behavior for jumping to a type is its declaration, which is here. Though this is better than ambient types in that we can continue to click to get to the implementation.

I tried defining the type at the site of its implementation, but defining it within a function (`start`) limits the scope of the typedef to within that function.